ddtrace/profiling/collector/memalloc.py

```diff
@@ -109,10 +109,9 @@ def _start_service(self):
 
         super(MemoryCollector, self)._start_service()
 
-    def _stop_service(self):
-        # type: (...) -> None
-        super(MemoryCollector, self)._stop_service()
-
+    @staticmethod
+    def on_shutdown():
```

Comment:
Unless I missed something, this PR is not calling the `memalloc.stop()` on `stop()` anymore and only on shutdown - do we expect the `MemoryCollector` to be stopped without being shut down?

Reply:
`on_shutdown` is called whenever a periodic service is stopped. So stopping the collector will eventually call `on_shutdown`. The extra guarantee here is that it will be called _after_ the periodic function, which is the one that potentially requires `memalloc` to be in the "started" state.